### PR TITLE
Test: Remove unused function

### PR DESCRIPTION
**Pull request by mmenbawy**
* [ ] I added CHANGELOG entry for this change.
* [x] Change is not relevant to the end user.

## Changes

- Remove unused function `func newMockAlertmanager` from test/e2e/rule_test.go
- Remove unused function `func targetAndAssert` from test/e2e/targets_api_test.go

### CODE REVIEW

**Review by squat — APPROVED** [review]
nice!

**Comment by squat** [comment]
Hi @Mostafa-Elmenbawy thank for this! plz sign the DCO (git commit --amend --signoff) and we can merge :)

**Comment by Namanl2001** [comment]
@Mostafa-Elmenbawy you also need to run `make go-lint` in the project root and commit the changes. Please see this failing [check](https://github.com/thanos-io/thanos/pull/4165/checks?check_run_id=2500167861)
Also don't forget to sign-off each of your commits to pass the DCO check. Thanks

File: test/e2e/targets_api_test.go [hunk 1]
```diff
@@ -106,40 +106,3 @@ func TestTargetsAPI_Fanout(t *testing.T) {
 		},
 	})
 }
-
-//nolint:unused
-func targetAndAssert(t *testing.T, ctx context.Context, addr string, state string, want *targetspb.TargetDiscovery) {
```
Comment (yeya24) [inline test/e2e/targets_api_test.go:111]: No, please keep it. This function is not used because we disabled the flaky test case for now.

Comment (yeya24) [reply test/e2e/targets_api_test.go:111]: See also https://github.com/thanos-io/thanos/pull/4153

**Review by yeya24 — CHANGES_REQUESTED** [review]

File: test/e2e/targets_api_test.go (hunk 1 above)
Comment (bwplotka) [reply test/e2e/targets_api_test.go:111]: 👍🏽

**Pull request closed without merging**